censys-subdomain-finder.py

```diff
@@ -10,95 +14,131 @@
 
 load_dotenv()
 
-NON_COMMERCIAL_API_LIMIT = 1000
+USER_AGENT = f"{CensysCerts.DEFAULT_USER_AGENT} (censys-subdomain-finder; +https://github.com/christophetd/censys-subdomain-finder)"
+
 
 # Finds subdomains of a domain using Censys API
 def find_subdomains(domain, api_id, api_secret, limit_results):
     try:
-        censys_certificates = CensysCertificates(api_id=api_id, api_secret=api_secret)
-        certificate_query = 'parsed.names: %s' % domain
+        censys_certificates = CensysCerts(
+            api_id=api_id, api_secret=api_secret, user_agent=USER_AGENT
+        )
+        certificate_query = "names: %s" % domain
         if limit_results:
-            certificates_search_results = censys_certificates.search(certificate_query, fields=['parsed.names'], max_records=NON_COMMERCIAL_API_LIMIT)
+            certificates_search_results = censys_certificates.search(
+                certificate_query, fields=["names"], per_page=100, pages=10
+            )
         else:
-            certificates_search_results = censys_certificates.search(certificate_query, fields=['parsed.names'])
+            certificates_search_results = censys_certificates.search(
+                certificate_query, fields=["names"]
+            )
 
         # Flatten the result, and remove duplicates
         subdomains = []
-        for search_result in certificates_search_results:
-            subdomains.extend(search_result['parsed.names'])
+        for page in certificates_search_results:
+            for search_result in page:
+                subdomains.extend(search_result["names"])
 
         return set(subdomains)
-    except censys.common.exceptions.CensysUnauthorizedException:
-        sys.stderr.write('[-] Your Censys credentials look invalid.\n')
+    except CensysUnauthorizedException:
+        sys.stderr.write("[-] Your Censys credentials look invalid.\n")
         exit(1)
-    except censys.common.exceptions.CensysRateLimitExceededException:
-        sys.stderr.write('[-] Looks like you exceeded your Censys account limits rate. Exiting\n')
+    except CensysRateLimitExceededException:
+        sys.stderr.write(
+            "[-] Looks like you exceeded your Censys account limits rate. Exiting\n"
+        )
         return set(subdomains)
-    except censys.common.exceptions.CensysException as e:
+    except CensysException as e:
         # catch the Censys Base exception, example "only 1000 first results are available"
-        sys.stderr.write('[-] Something bad happened, ' + repr(e))
+        sys.stderr.write("[-] Something bad happened, " + repr(e))
```

Comment:
So I tested on `main` (worked properly), and on the PR branch I now get:

```
[*] Applying non-commercial limits (1000 results at most)
[*] Searching Censys for subdomains of christophetd.fr
[-] Did not find any subdomain
[-] Something bad happened, 422 (Unprocessable Entity): argument of type 'NoneType' is not iterable
```

Reply:
Ah good call out, I am waiting for Censys to fix an issue with one of the endpoints: https://github.com/censys/censys-python/issues/475 we should wait to merge this until its fixed.

Reply:
It also only seems to happen when you specify fields as a community users so it should only affect this project, not CloudFlair.

Reply:
Okay, fixed in: 7920b5e